lib/utils.py

```diff
@@ -54,16 +54,19 @@ def addr_dns(octets):
 
 
 def ip_mib_address(key, item):
-    key = tuple(map(int, key.split('.')))
-    try:
-        local_typ = key[0]
-        local_typ_name, local_typ_func = ADDRESS_TP[local_typ]
-        local_addr = local_typ_func(key[1:])
-    except Exception:
-        raise ParseKeyException
+    if 'Addr' not in item:
+        # some devices return ipAddressAddr as a value so we don't have to
+        # derive it from the key
+        key = tuple(map(int, key.split('.')))
+        try:
+            local_typ = key[0]
+            local_typ_name, local_typ_func = ADDRESS_TP[local_typ]
+            local_addr = local_typ_func(key[1:])
+        except Exception:
+            raise ParseKeyException
 
-    item['AddrType'] = local_typ_name
-    item['Addr'] = local_addr
+        item['AddrType'] = local_typ_name
```

Comment:
@Koos85 , is AddrType already in `item` when `Addr` is in `item`? Maybe add this to the comment as this is not clear.